packages/client/src/lib.js

```diff
@@ -97,8 +99,38 @@ class Web3Storage {
     wrapWithDirectory = true,
     name
   } = {}) {
-    const url = new URL('/car', endpoint)
+    const blockstore = new Blockstore()
+
+    const { out, root } = await pack({
+      input: Array.from(files).map((f) => ({
+        path: f.name,
+        content: f.stream()
+      })),
+      blockstore,
+      wrapWithDirectory,
+      maxChunkSize: 1048576,
+      maxChildrenPerNode: 1024
+    })
+    /** @type {string} */
+    const carRoot = root.toString()
+
+    onRootCidReady && onRootCidReady(carRoot)
+    const car = await CarReader.fromIterable(out)
+    return this._put({ endpoint, token }, { onStoredChunk, car, maxRetries, blockstore, name })
+  }
+
+  /**
+   * @param {Service} service
+   * @param {_PutOptions} options
+   * @returns {Promise<CIDString>}
+   */
+  static async _put ({ endpoint, token }, {
+    onStoredChunk, car, name, blockstore,
+    maxRetries = MAX_PUT_RETRIES
+  }) {
```

Comment:
Why not call this `putCar`? I'd also pull the car out of the options object and have it as a separate parameter since it's required - then the only difference with the instance method is the credentials, and this is consistent with other static methods.

```suggestion
  /**
   * @param {Service} service
   * @param {import('@ipld/car/api').CarReader} car
   * @param {PutCarOptions} options
   * @returns {Promise<CIDString>}
   */
  static async putCar ({ endpoint, token }, car, {
    onStoredChunk, name, blockstore,
    maxRetries = MAX_PUT_RETRIES
  }) {
```

Reply:
@alanshaw I understand why you're suggesting this name change, but could you help me understand why it would be okay to have both this renamed `status async putCar` function (defined here) and a `putCar` function (defined on line 346) in the same file? Do they not need different names from each other?

Reply:
Static functions exist on the class (function) object:

```js
const cid = Web3Storage.putCar(...)
```

Instance methods exist on the prototype:

```js
const client = new Web3Storage()
const cid = client.putCar(...)
```

So there's no collision.

The `class` and `static` syntax is effectively sugar for:

```js
function Web3Storage {}
Web3Storage.putCar = async function () {}
Web3Storage.prototype.putCar = async function () {}
```